### PR TITLE
Enable Syntax Highlight and automatic formatting of JSON and XML

### LINKED ISSUE #5: Syntax highlighting
Basic syntax highlighting for various languages:

* JSON (obviously)
* XML

Decide on which library to use:

https://highlightjs.org
http://prismjs.com

### LINKED ISSUE #60: Feature: Auto-format/pretty-print XML
It's very common for webhooks to be delivered using XML.

Ideally, webhook.site should enable pretty-printing for XML to enable a better viewing/reading experience.

### PR DESCRIPTION
**Pull request by micsco**
Adds support for syntax highlighting of XML and JSON files (#5)

- Just JS/JSON and XML support enabled for now (to optimise JS size)
- Does a shallow copy of the currentRequest (so that when changes are made the original is not overriden)
- Enable formatting in the setCurrentRequest method
- Adds XML pretty printing/formatting as well

Fixes #5 
Fixes #60

### CODE REVIEW

**Comment by micsco** [comment]
@fredsted This works as I test it:
XML:
image

JSON:
image

**Comment by micsco** [comment]
Another JSON example with more types to show highlighting:
image

File: resources/assets/sass/app.scss [hunk 1]
```diff
@@ -1,4 +1,5 @@
 @import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap";
+@import "node_modules/highlight.js/styles/github";
```
Comment (micsco) [inline resources/assets/sass/app.scss]: @fredsted Are you happy with this as default style? Not sure what you wanted to use here; being able to choose would be great but added a level of complexity (plus additional preference to have somewhere)

I personally prefer a dark style

**Comment by micsco** [comment]
Added XML formatting too:

image

**Comment by fredsted** [comment]
Looks awesome! Great work. I think a lot of users will be happy about the XML formatting.

Regarding the theme, I'd like to keep it a bright theme so it fits with the rest of the app. I think an app-wide dark mode toggle would be nice.

**Comment by micsco** [comment]
Awesome, I haven't had a chance to test the limits of the formatters, but if you're happy with it go ahead and merge 👍

**Comment by fredsted** [comment]
Gave it a spin, and the only minor thing is that the formatting can't be toggled on/off immediately on the checkbox - if I want to see how the request looks unformatted, I have to turn formatting off and click back to another request. For me that's useful when you want to see *exactly* what's returned from the webhook. 

I might try to fix this myself, but going to merge this for now. 

Thanks again!

**Pull request merged**